modules/serverless-functions-automatic-build-deploy.adoc

```diff
@@ -0,0 +1,90 @@
+// Module included in the following assemblies:
+//
+// * functions/serverless-functions-on-cluster-builds.adoc
+
+:_content-type: PROCEDURE
+[id="serverless-functions-automatic-build-deploy_{context}"]
+= Configuring automatic building and deployment on GitHub Updates
+
+You can use the Shipwright framework to configure a function to be automatically built and deployed on the cluster whenever the source code in its GitHub repository is updated.
+
+.Prerequisites
+
+* {pipelines-title} must be installed on your cluster.
+
+* You have installed the OpenShift CLI (`oc`).
+
+* You have installed the Knative (`kn`) CLI.
+
+* Access to the GitHub repository containing your function's source code.
+
+.Procedure
+
+. Update the `func.yaml` file in your function project to include the GitHub repository URL and the branch to watch for updates:
++
+[source,yaml]
+----
+name: my-function
+namespace: ""
+runtime: node
+...
+repository: https://github.com/john-smith/my-repository.git <1>
+branch: main <2>
+----
+<1> Repository URL
+<2> Branch to monitor
+
+. Define the build configuration using Shipwright in your project. Create a `build.yaml` file:
```

Comment:
You've got two steps here in a single step. `Create a build.yaml file using Shipwright in your project to define the build configuration such as the following:` Something like that might work.